dms/models/dms_security_mixin.py

```diff
@@ -135,6 +136,22 @@ def _apply_access_groups(self, query, mode="read"):
             )
         query.where_clause += [where_clause]
         query.where_clause_params += [self.env.user.id]
+        # Add _get_directory_ids_with_res_model_without_access
+        if self.env.context.get("use_res_model_without_access", True):
+            custom_ids = self._get_directory_ids_with_res_model_without_access(mode)
+            if len(custom_ids) > 0:
+                field = "id"
+                if self._name == "dms.file":
+                    field = "directory_id"
+                extra_clause = """
+                    "{table}".{field} NOT IN ({ids})
+                """.format(
+                    table=self._table,
+                    field=field,
+                    ids=", ".join(["%s"] * len(custom_ids)) % tuple(custom_ids),
+                )
+                where_clause = "{extra_clause}".format(extra_clause=extra_clause)
+                query.where_clause += [where_clause]
```

Comment:
This would complement the comment above (and fix one little nonsense you can see here BTW):

```suggestion
                query.where_clause += [extra_clause]
                query.where_clause_params += custom_ids
```